lib/ansible/modules/cloud/amazon/lambda_info.py

```diff
@@ -389,7 +396,10 @@ def main():
     this_module_function = getattr(this_module, invocations[module.params['query']])
     all_facts = fix_return(this_module_function(client, module))
 
-    results = dict(ansible_facts={'lambda_facts': {'function': all_facts}}, changed=False)
+    if is_old_facts:
+        results = dict(ansible_facts={'lambda_facts': {'function': all_facts}}, changed=False)
+    else:
+        results = dict(lambda_facts={'function': all_facts}, changed=False)
```

Comment:
Renaming the module `lambda_info` and then returning `lambda_facts` just seems wrong. 

This is the best chance to change this to just return `lambda`.

Reply:
The problem is that the return value documented by the module is called `lambda_facts`. I don't see how to work around this without modifying the way how `RETURN` is parsed and interpreted.

So either don't rename the module; then we can deprecate the module and create a new `lambda_info` which doesn't return `lambda_facts`, or we keep the `lambda_facts` in there.

Or am I missing something?

Reply:
I've taken the deprecation/new module approach in 5e2f4695ad3af42eda288891c6f09e97f8bb35b9. The main disadvantage is that the old `_facts` module will no longer be tested (assuming it had tests before; which is not the case here).